crates/witness-generator/src/eest_generator.rs

```diff
@@ -0,0 +1,196 @@
+use anyhow::{Result, anyhow, bail};
+use async_trait::async_trait;
+use ef_tests::{
+    Case,
+    cases::blockchain_test::{BlockchainTestCase, run_case},
+    models::BlockchainTest,
+};
+use rayon::prelude::*;
+use std::{
+    path::{Path, PathBuf},
+    process::Command,
+    str::FromStr,
+};
+use walkdir::{DirEntry, WalkDir};
+
+use crate::{BlocksAndWitnesses, blocks_and_witnesses::WitnessGenerator};
+use reth_stateless::StatelessInput;
+
+/// Witness generator that produces `BlocksAndWitnesses` for execution-spec-test fixtures.
+#[derive(Debug, Clone)]
+pub struct ExecSpecTestBlocksAndWitnessBuilder {
+    tag: Option<String>,
+    include: Option<Vec<String>>,
+    exclude: Option<Vec<String>>,
+}
+
+impl ExecSpecTestBlocksAndWitnessBuilder {
+    /// Creates a new `ExecSpecTestBlocksAndWitnessBuilder` with default values.
+    pub fn new() -> Self {
+        Self {
+            tag: None,
+            include: None,
+            exclude: None,
+        }
+    }
+
+    /// Sets the tag for the execution-spec-test fixtures.
+    pub fn with_tag(mut self, tag: String) -> Self {
+        self.tag = Some(tag);
+        self
+    }
+
+    /// Includes only test names that contain the provided strings.
+    pub fn with_includes(mut self, includes: Vec<String>) -> Self {
+        self.include = Some(includes);
+        self
+    }
+
+    /// Excludes all test names that contain the provided strings.
+    pub fn with_excludes(mut self, exclude: Vec<String>) -> Self {
+        self.exclude = Some(exclude);
+        self
+    }
+
+    /// Builds the `ExecSpecTestBlocksAndWitnesses` instance.
+    pub fn build(self) -> Result<ExecSpecTestBlocksAndWitnesses> {
+        let tag = self.tag;
+        let include = self.include.unwrap_or_default();
+        let exclude = self.exclude.unwrap_or_default();
+
+        let mut cmd = Command::new("./scripts/download-and-extract-fixtures.sh");
+        if let Some(tag) = tag {
+            cmd.arg(tag);
+        }
+        let output = cmd.output()?;
+
+        if !output.status.success() {
+            bail!(
+                "Failed to download EEST benchmark fixtures: {}",
+                String::from_utf8_lossy(&output.stderr)
+            );
+        }
```

Comment:
I was very tempted to fully port the script to Rust, but this PR was already big.

Also, is that something we want, or we still find useful to have it as a bash script?
LMK your thoughts -- my take is prob makes sense to eventually port it to Rust to avoid this.